piker/brokers/kraken.py

```diff
@@ -313,29 +305,41 @@ async def submit_limit(
         price: float,
         action: str,
         size: float,
-        reqid: int = None,
-    ) -> int:
+        reqid: str = None,
+    ) -> dict:
         '''
         Place an order and return integer request id provided by client.
 
         '''
-        # Build order data for kraken api
-        data = {
-            "userref": reqid,
-            "ordertype": "limit",
-            "type": action,
-            "volume": str(size),
-            "pair": symbol,
-            "price": str(price),
-            # set to True test AddOrder call without a real submission
-            "validate": False
-        }
-        return await self.endpoint('AddOrder', data)
+        if reqid is None:
+            # Build order data for kraken api
+            data = {
+                "ordertype": "limit",
+                "type": action,
+                "volume": str(size),
+                "pair": symbol,
+                "price": str(price),
+                # set to True test AddOrder call without a real submission
+                "validate": False
+            }
+            return await self.endpoint('AddOrder', data)
+        else:
+            # Edit order data for kraken api
+            data = {
+                "txid": reqid,
+                "pair": symbol,
+                "price": str(price),
+                # set to True test EditOrder call without a real submission
```

Comment:
Hm, can we expose this up through the method as kwarg too maybe?

Both endpoints support it so i think just a wholesale flag would be handy here if we try to implement a pseudo-paper engine no?

Reply:
not sure what you mean by this, can you elaborate?

Reply:
the commented parameter that seems to make this a paper-like request?

Reply:
yes I hear you, as an arg in the `submit_limit` method, on it